blackbox/docs/upgrading.txt

```diff
@@ -0,0 +1,106 @@
+.. highlight:: sh
+
+=========
+Upgrading
+=========
+
+Sometimes CrateDB introduces breaking changes in new versions.
+
+When upgrading to a version with breaking changes, there may be upgrade steps you need to take. When upgrading, please see the corresponding section on this page to for more instructions.
+
+Upgrading to 1.2
+================
+
+To upgrade to CrateDB 1.2 you must first upgrade to 1.1.
+
+CrateDB 1.2 ships with Elasticsearch 5. This is big improvement for CrateDB, but as a result of changes in Elasticsearch 5, some of your tables may need to be migrated or reindexed. To help you with this, CrateDB 1.1 includes a number of cluster checks that must be run before you upgrade to 1.2.
+
+Once you have upgraded to CrateDB 1.1 and all of your cluster checks pass, you can safely upgrade to 1.2.
+
+Upgrading to 1.1
+================
+
+CrateDB 1.1 includes new cluster checks that will inform you about tables that need migrating or reindexing before upgrading to 1.2. CrateDB 1.1 will be the last release capable of reading these old tables.
+
+CrateDB 1.1 also includes a migration tool you can use to migrate your tables.
+
+To get started, upgrade your cluster like normal. When CrateDB 1.1 is running, you may now see that your cluster status is yellow rather than green.
+
+.. TODO: screenshot
+
+Depending on your situation, you may then be told that you either need to migrate or reindex some of your tables.
+
+.. TODO: screenshot
+
+.. TODO: screenshot
+
+If your cluster status is green, there is nothing more for you to do and you may safely upgrade to 1.2. Otherwise, follow the instructions below as instructed. Once you have completed the instructions, your cluster status should change to green, and you can safely upgrade to 1.2.
+
+As part of the upgrading process you will create a new snapshot for every table. Some snapshots created prior to CrateDB 1.1 will not work with 1.2.
```

Comment:
I'm not sure if this should be here or above where I mentioned... @seut What do you think?